src/core/src/descriptor/tensor.cpp

```diff
@@ -62,19 +60,13 @@ void ov::descriptor::Tensor::set_value_symbol(const TensorSymbol& value_symbol)
 }
 
 const ov::Shape& ov::descriptor::Tensor::get_shape() const {
-    if (m_partial_shape.is_static()) {
-        if (m_shape_changed.load(std::memory_order_relaxed)) {
-            std::lock_guard<std::mutex> guard(m_mutex);
-            if (m_shape_changed)  // double check after mutex lock
-            {
-                m_shape = m_partial_shape.to_shape();
-                m_shape_changed = false;
-            }
-        }
-        return m_shape;
-    } else {
-        throw std::invalid_argument("get_shape was called on a descriptor::Tensor with dynamic shape");
+    OPENVINO_ASSERT(m_partial_shape.is_static(), "get_shape was called on a descriptor::Tensor with dynamic shape");
```

Comment:
The line:
`m_shape = m_partial_shape.to_shape();`
should throw is convert to shape from dynamic, otherwise the `m_shape` should be set correctly. I think this  assert can be removed.

Reply:
Right. Removed redundant check.

Reply:
It looks like it has got impact on tests.

Reply:
Fixed